features/lorawan/LoRaWANStack.cpp

```diff
@@ -292,12 +319,32 @@ int16_t LoRaWANStack::handle_tx(const uint8_t port, const uint8_t* data,
         return status;
     }
 
-    if (flags == 0 ||
-        (flags & MSG_FLAG_MASK) == (MSG_CONFIRMED_FLAG|MSG_UNCONFIRMED_FLAG)) {
+    // case 1: no flag set
+    if (flags == 0) {
+        tr_error("CONFIRMED|UNCONFIRMED|PROPRIETARY flag must be set");
+        return LORAWAN_STATUS_PARAMETER_INVALID;
+    }
+
+    // case 2: check if both CONFIRMED and UNCONFIRMED flags set
+    if ((flags & ERR_FLAG_MUTUAL_EXCLUSION) == ERR_FLAG_MUTUAL_EXCLUSION) {
```

Comment:
What if neither are set (and it's a normal message)? Seems like this might be simpler if confirmed/unconfirmed was a single flag.

This is all very wordy to do a very simple test. I'd be vaguely inclined to just list out the valid values thus:

    switch (flags) {
         case UNCONFIRMED:
         case CONFIRMED:
         case PROPRIETARY:
              break;
      
         default:
              tr_error("Invalid send flags");
              return LORAWAN_STATUS_PARAMETER_INVALID;
     }

Reply:
case 1 handles if nothing is set. Multicast flag cannot be set for uplink. And all the flags are mutually exclusive, so they can't be used in pairs for uplink.